src/logic/safe/store/reducer/pendingTransactions.ts

```diff
@@ -46,6 +46,12 @@ export const pendingTransactionsReducer = handleActions<PendingTransactionsState
       // Omit id from the pending transactions on current chain
       const { [id]: _, ...newChainState } = state[chainId] || {}
 
+      if (Object.keys(newChainState[chainId] || {}).length === 0) {
+        // Omit chainId from the pending transactions if no pending transactions on chain
+        const { [chainId]: _, ...newState } = state
+        return newState
+      }
```

Comment:
Does omiting the chainId mean that we remove the entry if there are no pending txs for the current chain?

Reply:
Exactly. If the `chainId` isn't removed then we have an empty pending object.